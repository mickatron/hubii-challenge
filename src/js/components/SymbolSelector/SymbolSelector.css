.SymbolSelector {
  display: flex;
  margin: 0; 
  padding: 0;
}

.SymbolSelector li {
  width: 50%;
  list-style: none;
  margin: 0; 
  padding: 0;
}
.SymbolSelector__button {
  width: 100%;
  background: #eee;
  border: none;
  border-bottom: 1px solid #ccc;
  outline: none;
  height: 32px;
  cursor: pointer;
}

.SymbolSelector__button--active {
  border: 1px solid #ccc;
  border-bottom: none;
  background: none;
}