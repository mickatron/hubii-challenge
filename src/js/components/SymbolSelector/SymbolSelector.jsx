import classnames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';

import classNameType from 'js/propTypes/className';

import './SymbolSelector.css';

export default class SymbolSelector extends React.PureComponent {
  static propTypes = {
    activeSymbol: PropTypes.string,
    className: classNameType,
    symbols: PropTypes.arrayOf(PropTypes.string).isRequired,
    onClick: PropTypes.func.isRequired,
  };

  static defaultProps = {
    activeSymbol: undefined,
    className: undefined,
  }

  render() {
    const {
      activeSymbol,
      className,
      onClick,
      symbols,
    } = this.props;
    return (
      <ul className={classnames(className, 'SymbolSelector')}>
        {symbols.map(symbol => (
          <li key={symbol}>
            <button
              className={classnames(
                'SymbolSelector__button',
                { 'SymbolSelector__button--active': symbol === activeSymbol },
              )}
              onClick={onClick}
              value={symbol}
            >
              {symbol.replace('_', '/')}
            </button>
          </li>
        ))}
      </ul>
    );
  }
}
