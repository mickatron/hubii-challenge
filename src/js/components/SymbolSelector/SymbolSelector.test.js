import React from 'react';
import { shallow } from 'enzyme';

import SymbolSelector from './SymbolSelector';

describe('SymbolSelector', () => {
  let component = null;
  describe('renders', () => {
    it('a snapshot', () => {
      component = shallow((
        <SymbolSelector
          activeSymbol="BTC_USD"
          symbols={[
            'BTC_USD',
            'ETH_USD',
          ]}
          onClick={() => {}}
        />
      ));
      expect(component).toMatchSnapshot();
    });
  });
});
