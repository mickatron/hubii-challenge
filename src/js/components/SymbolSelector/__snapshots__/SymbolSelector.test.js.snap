// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SymbolSelector renders a snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SymbolSelector
    activeSymbol="BTC_USD"
    className={undefined}
    onClick={[Function]}
    symbols={
        Array [
            "BTC_USD",
            "ETH_USD",
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li>
          <button
                    className="SymbolSelector__button SymbolSelector__button--active"
                    onClick={[Function]}
                    value="BTC_USD"
          >
                    BTC/USD
          </button>
</li>,
        <li>
          <button
                    className="SymbolSelector__button"
                    onClick={[Function]}
                    value="ETH_USD"
          >
                    ETH/USD
          </button>
</li>,
      ],
      "className": "SymbolSelector",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "BTC_USD",
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="SymbolSelector__button SymbolSelector__button--active"
            onClick={[Function]}
            value="BTC_USD"
>
            BTC/USD
</button>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "BTC/USD",
            "className": "SymbolSelector__button SymbolSelector__button--active",
            "onClick": [Function],
            "value": "BTC_USD",
          },
          "ref": null,
          "rendered": "BTC/USD",
          "type": "button",
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "ETH_USD",
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="SymbolSelector__button"
            onClick={[Function]}
            value="ETH_USD"
>
            ETH/USD
</button>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "ETH/USD",
            "className": "SymbolSelector__button",
            "onClick": [Function],
            "value": "ETH_USD",
          },
          "ref": null,
          "rendered": "ETH/USD",
          "type": "button",
        },
        "type": "li",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li>
            <button
                        className="SymbolSelector__button SymbolSelector__button--active"
                        onClick={[Function]}
                        value="BTC_USD"
            >
                        BTC/USD
            </button>
</li>,
          <li>
            <button
                        className="SymbolSelector__button"
                        onClick={[Function]}
                        value="ETH_USD"
            >
                        ETH/USD
            </button>
</li>,
        ],
        "className": "SymbolSelector",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "BTC_USD",
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="SymbolSelector__button SymbolSelector__button--active"
              onClick={[Function]}
              value="BTC_USD"
>
              BTC/USD
</button>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "BTC/USD",
              "className": "SymbolSelector__button SymbolSelector__button--active",
              "onClick": [Function],
              "value": "BTC_USD",
            },
            "ref": null,
            "rendered": "BTC/USD",
            "type": "button",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "ETH_USD",
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="SymbolSelector__button"
              onClick={[Function]}
              value="ETH_USD"
>
              ETH/USD
</button>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "ETH/USD",
              "className": "SymbolSelector__button",
              "onClick": [Function],
              "value": "ETH_USD",
            },
            "ref": null,
            "rendered": "ETH/USD",
            "type": "button",
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
